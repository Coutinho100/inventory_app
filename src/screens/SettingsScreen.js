import React from 'react';
import { View, Text, Platform } from 'react-native';
import HeaderButton from '../components/Component/HeaderButton';
import { HeaderButtons, Item } from 'react-navigation-header-buttons';
import Colors from '../constants/Colors'

const SettingsScreen = props => {

    return (
        <View>
            <Text>
                Settings Screen
            </Text>
        </View>
    );
};



SettingsScreen.navigationOptions = navData => {
    return {
        headerTitle: "Setting",
        headerStyle: {
            backgroundColor: Platform.OS === 'android' ? Colors.primaryColor : ''
        },
        headerTitleStyle: {
            fontFamily: 'open-sans'
        },
        headerLeft: () => (
            <HeaderButtons HeaderButtonComponent={HeaderButton} >
                <Item title="Menu" iconName="ios-menu" onPress={() => {
                    navData.navigation.toggleDrawer()
                }} />
            </HeaderButtons>
        )
    }
}

export default SettingsScreen;