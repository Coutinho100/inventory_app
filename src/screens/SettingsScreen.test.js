import React from 'react';
import SettingsScreen  from './SettingsScreen';
import renderer from 'react-test-renderer';

import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import test_function, {findByTestAttr, mockPlatform, findByTestProps, mockAlert} from '../_test_/componenets/test_function';
// import Alert from 'Alert';

// import {act} from 'react-dom/test-utils';
// JestHook.mock('expo-font');
import configureStore from 'redux-mock-store';
import TestRenderer from 'react-test-renderer';
import Contact from '../models/contact'
const {act} = TestRenderer;
const mockStore = configureStore([]);

 
configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()



const dummy_data = {

    navigation:{toggleDrawer:function_click, navigate:function_click}

}


describe('<SettingsScreen />', () => {

  
    
  it("Events and props test in SettingsScreen", () =>
        {
            const wrapper = shallow(<SettingsScreen {...dummy_data}></SettingsScreen>)
            expect(wrapper).toMatchSnapshot();
           const navOption= SettingsScreen.navigationOptions(dummy_data)
           let wrap= shallow(navOption.headerLeft())
            let toggle = findByTestAttr(wrap, "navToggle")
            expect(toggle).toHaveLength(1)
            toggle.props().onPress()
            expect(function_click).toHaveBeenCalledTimes(1)
        });

  
        });


