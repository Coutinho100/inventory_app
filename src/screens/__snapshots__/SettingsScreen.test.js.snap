// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> Events and props test in SettingsScreen 1`] = `ShallowWrapper {}`;
